src/components/ChatEntry.js

```diff
@@ -1,22 +1,35 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({id, sender, body, timeStamp, liked, updateLikes }) => {
+  const handleLike = () => {
+    updateLikes(id);
+  };
 
-const ChatEntry = (props) => {
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className="chat-entry">
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time">
+          <TimeStamp time={timeStamp} />
+        </p>
+        <button className={liked ? 'like liked' : 'like'} onClick={handleLike}>
+          {liked ? '❤️' : '🤍'}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  updateLikes: PropTypes.func.isRequired, 
```

Comment:
Because this proptype is marked `isRequired`, we're getting warnings in the tests, since the test author didn't know what you were going to call this property, and so it's absent from the props passed in during the tests.

Ideally, we'd also include code to ensure that any "optional" values are actually present before trying to use them. For example, in the like click handler we could do something like
```js
    if (updateLikes) {
        updateLikes(id);
    }
```
which would only try to call the function if it's not falsy (and `undefined` is falsy).

Reply:
👀 Note that in your props destructuring, you included `id` as an expected value to be passed down (and wrote code that purportedly uses it... more on that in ChatLog!) but didn't include it in the proptypes here. It turns out that your ChatLog is sidestepping the issue, meaning that it's not a problem that the id is `undefined`. But if you really don't want to use the id, then don't include code that destructures and uses it. If you _do_ want to use the id, be sure to include it in the proptypes.